### ISSUE TITLE
Review pyflakes warnings for Trial

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11198]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#1719|
|Type|task|
|Created|2006-05-15 09:49:41Z|

```
twisted/trial/reporter.py:13: 'types' imported but unused
twisted/trial/reporter.py:18: 'defer' imported but unused
twisted/trial/reporter.py:19: 'util' imported but unused
twisted/trial/runner.py:162: 'utils' imported but unused
twisted/trial/runner.py:382: 'readline' imported but unused
twisted/trial/unittest.py:7: 'errno' imported but unused
twisted/trial/unittest.py:7: 'time' imported but unused
twisted/trial/unittest.py:10: 'reflect' imported but unused
twisted/trial/unittest.py:555: undefined name 'WaitIsNotReentrantError'
twisted/trial/unittest.py:555: undefined name 'REENTRANT_WAIT_ERROR_MSG'
twisted/trial/unittest.py:616: 'reporter' imported but unused
twisted/trial/util.py:22: 'warnings' imported but unused
twisted/trial/util.py:22: 'signal' imported but unused
twisted/trial/util.py:22: 'time' imported but unused
twisted/trial/util.py:23: 'util' imported but unused
twisted/trial/util.py:23: 'failure' imported but unused
twisted/trial/util.py:24: 'defer' imported but unused
twisted/trial/util.py:165: undefined name 'timedRun'
twisted/trial/test/test_deferred.py:1: 'StringIO' imported but unused
twisted/trial/test/test_doctest.py:9: 'sys' imported but unused
twisted/trial/test/test_doctest.py:9: 'doctest' imported but unused
twisted/trial/test/test_doctest.py:9: 'exceptions' imported but unused
twisted/trial/test/test_doctest.py:11: 'trial' imported but unused
twisted/trial/test/test_doctest.py:13: 'itrial' imported but unused
twisted/trial/test/test_doctest.py:14: 'failure' imported but unused
twisted/trial/test/test_doctest.py:18: 'pprint' imported but unused
twisted/trial/test/test_doctest.py:20: 'zi' imported but unused
twisted/trial/test/test_loader.py:4: 'unittest' imported but unused
twisted/trial/test/test_output.py:2: 'unittest' imported but unused
twisted/trial/test/test_output.py:4: 'util' imported but unused
twisted/trial/test/test_output.py:5: 're' imported but unused
twisted/trial/test/test_reporter.py:7: 'time' imported but unused
twisted/trial/test/test_reporter.py:70: 'sys' imported but unused
twisted/trial/test/test_runner.py:91: 'TrialRunner' imported but unused
twisted/trial/test/test_runner.py:264: 'TrialSuite' imported but unused
twisted/trial/test/test_test_visitor.py:25: 'TestVisitor' imported but unused
twisted/trial/test/test_tests.py:88: undefined name 'SkipTest'
twisted/trial/test/test_tests.py:138: undefined name 'SkipTest'
twisted/trial/test/test_util.py:1: 'log' imported but unused
twisted/trial/test/test_util.py:2: 'utils' imported but unused
twisted/trial/test/test_util.py:3: 'defer' imported but unused
twisted/trial/test/test_util.py:3: 'interfaces' imported but unused
twisted/trial/test/test_util.py:3: 'threads' imported but unused
twisted/trial/test/test_util.py:3: 'reactor' imported but unused
twisted/trial/test/test_util.py:7: 'time' imported but unused
```

<details><summary>Searchable metadata</summary>

```
trac-id__1719 1719
type__task task
reporter__jml jml
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1147686581000000 1147686581000000
changetime__1147746356000000 1147746356000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @radix|
|-|-|

All done.  

I'm not sure about the 'readline' import.   The TrialRunner, TrialSuite and TestVisitor imports are all valid, if a little over-zealous.

Picking someone at random.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

To reproduce:

```
$ pyflakes twisted/trial
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @jerub|
|-|-|

The branch is flakes-1719

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)|@jerub set owner to @jml|
|-|-|

Excellent, I love things that pyflakes (almost) clean. Please merge.

for reference:
```
~/src/Twisted/trunk stephen$ pyflakes twisted/trial
twisted/trial/runner.py:382: 'readline' imported but unused
twisted/trial/test/test_runner.py:91: 'TrialRunner' imported but unused
twisted/trial/test/test_runner.py:264: 'TrialSuite' imported but unused
twisted/trial/test/test_test_visitor.py:25: 'TestVisitor' imported but unused
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set status to `closed`|
|-|-|

(In [16834]) Clean up python flakes.

 * Reviewer: jerub
 * Author: jml
 * Fixes [#11198](https://github.com/twisted/twisted/issues/11198)
 * Merges [branches/flakes-1719](branches/flakes-1719)

Clean up unused imports and a few obvious mistakes (referring to non-existent
names).  Not entirely pyflakes-clean, because some of our tests simply check
that things can be imported.

NB: This commit removes `t.trial.util.testFunction`.  It was never documented
and didn't really have any use outside of Trial.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
